worlds/earthbound/Client.py

```diff
@@ -0,0 +1,504 @@
+                    "operations": [{"operation": "replace", "value": int.from_bytes(melody_table, "little")}]
+                }])
+
+        await ctx.send_msgs([{
+                    "cmd": "Set",
+                    "key": f"{ctx.team}_{ctx.slot}_earthpower",
+                    "default": None,
+                    "want_reply": True,
+                    "operations": [{"operation": "replace", "value": int.from_bytes(earth_power_absorbed, "little")}]
+                }])
+
+        # death link handling goes here
+        if "DeathLink" in ctx.tags and ctx.last_death_link + 1 < time.time():
+            send_deathlink = await snes_read(ctx, PLAYER_JUST_DIED_SEND_DEATHLINK, 1)
+            currently_dead = send_deathlink[0] != 0x00
+            if send_deathlink[0] != 0x00:
+                snes_buffered_write(ctx, PLAYER_JUST_DIED_SEND_DEATHLINK, bytes([0x00]))
+            await ctx.handle_deathlink_state(currently_dead)
+
+        new_checks = []
+        from .game_data.local_data import check_table
+
+        location_ram_data = await snes_read(ctx, WRAM_START + 0x9C00, 0x88)
+        shop_location_flags = await snes_read(ctx, WRAM_START + 0xB721, 0x41)
+        for loc_id, loc_data in check_table.items():
+            if loc_id not in ctx.locations_checked:
+                if loc_id >= 0xEB1000:
+                    data = shop_location_flags[loc_data[0]]
+                else:
+                    data = location_ram_data[loc_data[0]]
+                masked_data = data & (1 << loc_data[1])
+                bit_set = masked_data != 0
+                invert_bit = ((len(loc_data) >= 3) and loc_data[2])
+                if bit_set != invert_bit and loc_id in ctx.server_locations:
+                    if text_open[0] == 0xFF or shop_scout[0]:  # Don't check locations while in a textbox
+                        new_checks.append(loc_id)
+                        
+        for new_check_id in new_checks:
+            ctx.locations_checked.add(new_check_id)
+            location = ctx.location_names.lookup_in_slot(new_check_id)
+            snes_logger.info(
+                f'New Check: {location} ({len(ctx.locations_checked)}/{len(ctx.missing_locations) + len(ctx.checked_locations)})')
+            await ctx.send_msgs([{"cmd": 'LocationChecks', "locations": [new_check_id]}])
+            await snes_write(ctx, [(WRAM_START + 0x0770, bytes([0]))])
+
+        if item_received[0] or special_received[0] != 0x00 or money_received[0] != 0x00:  # If processing any item from the server
+            return
+
+        is_energylink_enabled = await snes_read(ctx, IS_ENERGYLINK_ENABLED, 1)
+        is_requesting_energy = await snes_read(ctx, WRAM_START + 0x0790, 1)
+        energy_withdrawal = await snes_read(ctx, WRAM_START + 0x0796, 4)
+        ctx.set_notify(f"EnergyLink{ctx.team}")
+        energy = ctx.stored_data.get(f"EnergyLink{ctx.team}", 0)
+        exchange_rate = 1000000
+        if is_energylink_enabled[0]:
```

Comment:
I'm going to suggest being a bit careful with checks like these, as some wonky PC and/or connection to the SNES may return a `None` upon calling `snes_read` which could lead in your client disconnecting/crashing. Also, the more of these read requests you do, the higher the probabilities of them failing are and the slower the client becomes.

@beauxq was very kind to inform me about that in my now-defunct Mega Man X PR which I strongly suggest to read and apply such changes in your client:  https://github.com/ArchipelagoMW/Archipelago/pull/3842#discussion_r1729788590

tl;dr do more big RAM/ROM reads and handle `None` cases